crates/bindings-csharp/Runtime/Internal/ITable.cs

```diff
@@ -108,29 +100,27 @@ public IEnumerable<T> Parse()
                 using var reader = new BinaryReader(stream);
                 while (stream.Position < stream.Length)
                 {
-                    yield return IStructuralReadWrite.Read<T>(reader);
+                    yield return Read<T>(reader);
                 }
             }
         }
     }
+}
+
+public interface ITableView<View, T>
+    where View : ITableView<View, T>
+    where T : ITable<T>, new()
+{
+    static abstract T ReadGenFields(BinaryReader reader, T row);
+
+    // These are static helpers that codegen can use.
 
-    private class RawTableIter(FFI.TableId tableId) : RawTableIterBase
+    private class RawTableIter(FFI.TableId tableId) : ITable<T>.RawTableIterBase
     {
         protected override void IterStart(out FFI.RowIter handle) =>
             FFI.datastore_table_scan_bsatn(tableId, out handle);
     }
 
-    private class RawTableIterByColEq(FFI.TableId tableId, FFI.ColId colId, byte[] value)
-        : RawTableIterBase
-    {
-        protected override void IterStart(out FFI.RowIter handle)
-        {
-            _ = (tableId, colId, value);
-            // Needs to be implemented via datastore_table_scan_bsatn
-            throw new NotImplementedException();
-        }
-    }
-
     // Note: this must be Lazy to ensure that we don't try to get the tableId during startup, before the module is initialized.
     private static readonly Lazy<FFI.TableId> tableId_ =
```

Comment:
These no longer need to be Lazy. As the comment says, it was necessary only because we used static tables. Converting them to regular properties should make code a bit simpler.